scripts/tile.py

```diff
@@ -66,67 +66,77 @@ def tiler(stack, date, mgrs, bucket):
-
-                # Track band names and color interpretation
-                tile.attrs["long_name"] = [str(x.values) for x in tile.band]
-                color = [ColorInterp.blue, ColorInterp.green, ColorInterp.red] + [
-                    ColorInterp.gray
-                ] * (len(tile.band) - 3)
-
-                # Write tile to tempdir
-                name = "{dir}/claytile_{mgrs}_{date}_v{version}_{counter}.tif".format(
-                    dir=dir,
-                    mgrs=mgrs,
-                    date=date.replace("-", ""),
-                    version=VERSION,
-                    counter=str(counter).zfill(4),
-                )
-                tile.rio.to_raster(name, compress="deflate")
-
-                with rasterio.open(name, "r+") as rst:
-                    rst.colorinterp = color
-                    rst.update_tags(date=date)
-
-        print(f"Syncing {dir} with s3://{bucket}/{VERSION}/{mgrs}/{date}")
-        subprocess.run(
-            [
-                "aws",
-                "s3",
-                "sync",
-                dir,
-                f"s3://{bucket}/{VERSION}/{mgrs}/{date}",
-                "--no-progress",
-            ],
-            check=True,
-        )
+    # Iterate through each chunk of x and y dimensions and create tiles
+    for y_idx in range(num_y_tiles):
+        for x_idx in range(num_x_tiles):
+            # Calculate the start and end indices for x and y dimensions
+            # for the current tile
+            x_start = x_idx * TILE_SIZE
+            y_start = y_idx * TILE_SIZE
+            x_end = x_start + TILE_SIZE
+            y_end = y_start + TILE_SIZE
+
+            # Select the subset of data for the current tile
+            parts = [part[:, y_start:y_end, x_start:x_end] for part in stack]
+
+            # Only concat here to save memory, it converts S2 data to float
+            tile = xr.concat(parts, dim="band").rename("tile")
+
+            counter += 1
+            if counter % 100 == 0:
+                print(f"Counted {counter} tiles")
+
+            if not filter_clouds_nodata(tile):
+                continue
+
+            tile = tile.drop_sel(band="SCL")
+
+            # Track band names and color interpretation
+            tile.attrs["long_name"] = [str(x.values) for x in tile.band]
+            color = [ColorInterp.blue, ColorInterp.green, ColorInterp.red] + [
+                ColorInterp.gray
+            ] * (len(tile.band) - 3)
+
+            # Write tile to tempdir
+            name = "{dir}/claytile_{mgrs}_{date}_v{version}_{counter}.tif".format(
+                dir=dir,
+                mgrs=mgrs,
+                date=date.replace("-", ""),
+                version=VERSION,
+                counter=str(counter).zfill(4),
+            )
+            tile.rio.to_raster(name, compress="deflate")
+
+            with rasterio.open(name, "r+") as rst:
+                rst.colorinterp = color
+                rst.update_tags(date=date)
```

Comment:
Do you want to handle https://github.com/Clay-foundation/model/issues/70 here, i.e. use [ACQUISITIONDATETIME](https://gdal.org/user/raster_data_model.html#imagery-domain-remote-sensing) or [TIFFTAG_DATETIME](https://gdal.org/user/raster_data_model.html#metadata)? Or do that in a separate PR?

Reply:
Separate PR, coming up.